commcare_connect/templates/opportunity/invoice_list.html

```diff
@@ -0,0 +1,104 @@
+{% extends "opportunity/base.html" %}
+{% load django_tables2 %}
+{% load static %}
+{% load crispy_forms_tags %}
+
+{% block title %}{{ request.org }} - Invoices{% endblock title %}
+
+{% block breadcrumbs_inner %}
+  {{ block.super }}
+  <li class="breadcrumb-item">
+    <a href="{% url 'opportunity:detail' org_slug=request.org.slug pk=opportunity.pk %}">
+      {{ opportunity.name }}
+    </a>
+  </li>
+  <li class="breadcrumb-item active" aria-current="page">Invoices</li>
+{% endblock %}
+
+{% block content %}
+  <h2>Invoices</h2>
+  <hr />
+  {% if not request.org_membership.is_program_manager %}
+    <button type="button" class="btn btn-primary mb-2" data-bs-toggle="modal" data-bs-target="#invoiceModal">
+      Add New Invoice
+    </button>
+  {% endif %}
+
+  <form hx-get="{% url "opportunity:invoice_table" org_slug=request.org.slug pk=opportunity.pk %}"
+        hx-trigger="load, change, newInvoice from:body"
+        hx-target="#invoiceTable">
+    <select class="form-select mb-2" id="filterSelect" name="filter">
+      <option selected value="">Filter Invoices</option>
+      <option value="paid">Paid</option>
+      <option value="pending">Pending</option>
+    </select>
+  </form>
+
+  {% if request.org_membership.is_program_manager %}
+    <form x-data="{
+            selectAll: false,
+            selected: [],
+            toggleSelectAll() {
+              this.selectAll = !this.selectAll;
+              if(this.selectAll)
+                this.selected = JSON.parse(document.getElementById('invoice_ids').textContent);
+              else
+                this.selected = [];
+            }
+          }"
+      hx-post="{% url "opportunity:invoice_approve" org_slug=request.org.slug pk=opportunity.pk %}"
+      hx-swap="none"
+      hx-headers='{"X-CSRFToken": "{{ csrf_token }}"}'>
```

Comment:
why does this need a csrf token? shouldn't the django form handle this?

Reply:
Django was not able to find csrf token on forms submitted using 'hx-post'. This include the csrf token as a request header to provide the csrf token.